src/components/safe-apps/SafeAppLandingPage/UseApp.tsx

```diff
@@ -0,0 +1,110 @@
+import { Box, Button, MenuItem, Select, Typography, Grid, FormControl, InputLabel } from '@mui/material'
+import { ConnectedWallet } from '@/hooks/wallets/useOnboard'
+import { useState } from 'react'
+import { useAppSelector } from '@/store'
+import { selectAddressBookByChain } from '@/store/addressBookSlice'
+import { useLastSafe } from '@/hooks/useLastSafe'
+import { parsePrefixedAddress } from '@/utils/addresses'
+import SafeIcon from '@/components/common/SafeIcon'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import { AppRoutes } from '@/config/routes'
+import { CTA_BUTTON_WIDTH, CTA_HEIGHT } from '@/components/safe-apps/SafeAppLandingPage/constants'
+
+type Props = {
+  appUrl: string
+  wallet: ConnectedWallet | null
+  onConnectWallet: () => Promise<void>
+  safes: string[]
+  chainId: string
+  chainPrefix?: string
+}
+
+const UseApp = ({ wallet, onConnectWallet, safes, chainId, chainPrefix, appUrl }: Props): React.ReactElement => {
```

Comment:
I feel like this could be easily confused with a hook. I'd suggest `TryApp` or something similar instead.